GitHubBranches.ps1

```diff
@@ -1072,128 +1074,1140 @@ filter Remove-GitHubRepositoryBranchProtectionRule
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .INPUTS
+        GitHub.Branch
+        GitHub.Content
+        GitHub.Event
+        GitHub.Issue
+        GitHub.IssueComment
+        GitHub.Label
+        GitHub.Milestone
+        GitHub.PullRequest
+        GitHub.Project
+        GitHub.ProjectCard
+        GitHub.ProjectColumn
+        GitHub.Release
+        GitHub.Repository
+
+    .OUTPUTS
+        GitHub.BranchPatternProtectionRule
+
+    .EXAMPLE
+        Get-GitHubRepositoryBranchPatternProtectionRule -OwnerName microsoft -RepositoryName PowerShellForGitHub -BranchPatternName release/**/*
+
+        Retrieves branch protection rules for the release/**/* branch pattern of the PowerShellForGithub repository.
+
+    .EXAMPLE
+        Get-GitHubQlRepositoryBranchPatternProtectionRule -Uri 'https://github.com/microsoft/PowerShellForGitHub' -BranchPatternName master
+
+        Retrieves branch protection rules for the master branch pattern of the PowerShellForGithub repository.
+#>
+    [CmdletBinding(
+        PositionalBinding = $false,
+        DefaultParameterSetName = 'Elements')]
+    [OutputType({ $script:GitHubBranchProtectionRuleTypeName })]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSReviewUnusedParameter", "",
+        Justification="The Uri parameter is only referenced by Resolve-RepositoryElements which get access to it from the stack via Get-Variable -Scope 1.")]
+    param(
+        [Parameter(ParameterSetName = 'Elements')]
+        [string] $OwnerName,
+
+        [Parameter(ParameterSetName = 'Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            Position = 1,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'Uri')]
+        [Alias('RepositoryUrl')]
+        [string] $Uri,
+
+        [Parameter(
+            Mandatory,
+            Position = 2)]
+        [string] $BranchPatternName,
+
+        [string] $AccessToken
+    )
+
+    Write-InvocationLog
+
+    $elements = Resolve-RepositoryElements
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName' = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+    }
+
+    $branchProtectionRuleFields = 'allowsDeletions allowsForcePushes dismissesStaleReviews id ' +
+        'isAdminEnforced pattern requiredApprovingReviewCount requiredStatusCheckContexts ' +
+        'requiresApprovingReviews requiresCodeOwnerReviews requiresCommitSignatures requiresLinearHistory ' +
+        'requiresStatusChecks requiresStrictStatusChecks restrictsPushes restrictsReviewDismissals ' +
+        'pushAllowances(first: 100) { nodes { actor { ... on App { __typename name } ' +
```

Comment:
Done. I did think about coding the paging required to cope with more than 100 objects, but for the 0.0000001% of cases that might need it, I decided it wasn't worth it. It can be done in a future PR if anyone has the inclination.

Reply:
Agreed, and having it broken out to a constant makes that logic separation all the more clear for that potential future improvement.